packages/headless/src/controllers/search-parameter-manager/headless-search-parameter-manager.ts

```diff
@@ -45,6 +49,13 @@ const initialStateSchema = new Schema<
  * The `SearchParameterManager` controller allows restoring parameters that affect the results from e.g. a url.
  * */
 export interface SearchParameterManager extends Controller {
+  /**
+   * Updates the search parameters in state with the passed parameters and executes a search. Unspecified keys are reset to their initial values.
+   *
+   * @param parameters - The search parameters to synchronize.
+   */
+  synchronize(parameters: SearchParameters): void;
```

Comment:
The name `synchronize` confused me a bit until I read the doc about this function. Would `apply` be a good alternative?

Reply:
I prefer `synchronize` in this case because the function is doing two things:

1. It is applying the passed parameter values in state.
2. It is resetting the parameters that are not passed to their initial values.

If the method was called `apply` or `update`, I would only expect it to do (1) but not (2).